pkg/cluster/fixssh.go

```diff
@@ -56,52 +74,99 @@ func (m *manager) fixSSH(ctx context.Context) error {
 			if err != nil {
 				return err
 			}
 		}
 	}
+	return nil
+}
 
+func (m *manager) removeBackendPoolsFromNIC(ctx context.Context, resourceGroup, nicName string, nic *mgmtnetwork.Interface) error {
+	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
+		return fmt.Errorf("unable to remove backendpools from NIC as there are no IP configurations for %s in resource group %s", nicName, resourceGroup)
+	}
+	ipc := (*nic.InterfacePropertiesFormat.IPConfigurations)[0]
+	if ipc.LoadBalancerBackendAddressPools != nil {
+		m.log.Printf("Removing Loadbalancer Backend Address Pools from NIC %s with no VMs attached", nicName)
+		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = []mgmtnetwork.BackendAddressPool{}
+		return m.interfaces.CreateOrUpdateAndWait(ctx, resourceGroup, nicName, *nic)
+	}
 	return nil
 }
 
-func updateNIC(nic *mgmtnetwork.Interface, lb *mgmtnetwork.LoadBalancer, i int) bool {
-	id := fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)
+func (m *manager) updateNIC(ctx context.Context, nic *mgmtnetwork.Interface, nicName string, lb *mgmtnetwork.LoadBalancer, i int, infraID string) bool {
+	if nic.InterfacePropertiesFormat.IPConfigurations == nil || len(*nic.InterfacePropertiesFormat.IPConfigurations) == 0 {
+		m.log.Warnf("unable to update NIC as there are no IP configurations for %s", nicName)
+		return false
+	}
+
+	sshID := fmt.Sprintf("%s/backendAddressPools/ssh-%d", *lb.ID, i)
+	ilbID := fmt.Sprintf("%s/backendAddressPools/%s", *lb.ID, infraID)
+
+	updateSSHPool := true
+	updateILBPool := true
+
 	for _, p := range *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools {
-		if strings.EqualFold(*p.ID, id) {
-			return false
+		if strings.EqualFold(*p.ID, sshID) {
+			updateSSHPool = false
+		}
+		if strings.EqualFold(*p.ID, ilbID) {
+			updateILBPool = false
 		}
 	}
 
-	*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
-		ID: &id,
-	})
+	if updateSSHPool {
+		m.log.Printf("Adding NIC %s to Internal LoadBalancer SSH Backend Address Pool %s", nicName, sshID)
+		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
+			ID: &sshID,
+		})
+	}
+
+	if updateILBPool {
+		m.log.Printf("Adding NIC %s to Internal LoadBalancer API Address Pool %s", nicName, ilbID)
+		*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = append(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools, mgmtnetwork.BackendAddressPool{
+			ID: &ilbID,
+		})
+	}
 
-	return true
+	return updateSSHPool || updateILBPool
 }
 
-func updateLB(lb *mgmtnetwork.LoadBalancer) (changed bool) {
+func (m *manager) updateLB(ctx context.Context, lb *mgmtnetwork.LoadBalancer) (lbChanged bool) {
```

Comment:
All the changes to this function aren't required. If you were going to use the logger, then is would make sense to make it part of the manager interface, but since we aren't we shouldn't make any of these changes.